php/libraries/Candidate.class.inc

```diff
@@ -323,7 +325,23 @@ class Candidate implements \LORIS\StudyEntities\AccessibleResource
         // Return the new CandID
         return $candID;
     }
-
+    /**
+     * Generate a ParentID for a candidate
+     *
+     * @param string $IDtype  likes Parent
+     *
+     * @return string $randid likes Parent23456
+     * @throws DatabaseException
+     */
+    function _generateID(string $IDtype): string {
```

Comment:
perhaps a ParentIDGenerator class can be created that extends SiteIDGenerator or IdentifierGenerator class. and then in the createNew() method above, you can call `$parentID = (new \ParentIDGenerator())->generate();`. this would be in line with how CandIDs, PSCIDs and ExternalIDs are currently being generated.

Here's an example of how I did it for QPN (I copied the PSCIDGenerator and ExternalIDGenerator classes): https://github.com/aces/QPN/blob/master/libraries/PatientIDGenerator.php